### PR TITLE
Improve cunumeric.clip code coverage

### PR DESCRIPTION
**Pull request by yimoj**

### CODE REVIEW

File: tests/integration/test_clip.py [hunk 1]
```diff
@@ -140,6 +140,18 @@ def test_out_ndim():
     assert np.array_equal(out_np, out_num)
 
 
+def test_out_np_array():
+    array = ((2, 3, 4), (3, 4, 5), (6, 6, 12))
+    amin = (2, 3, 1)
+    amax = 6
+    np_arr = np.array(array)
+    num_arr = num.array(array)
+    out_np = out_num = np.zeros(np_arr.shape)
```
Comment (bryevdv) [inline tests/integration/test_clip.py:149]: This seems like a mistake. This only makes a single "out" array, and points both variable names at that same array:
```
In [1]: import numpy as np

In [2]: out_np = out_num = np.zeros(10)

In [3]: out_np is out_num
Out[3]: True
```
That means the comparison at the end is a self-comparison

File: tests/integration/test_clip.py [hunk 2]
```diff
@@ -172,6 +184,18 @@ def test_out(ndim):
     assert np.array_equal(out_np, out_num)
 
 
+def test_out_with_array_amin():
+    array = ((2, 3, 4), (3, 4, 5), (6, 6, 12))
+    amin = (2, 3, 1)
+    amax = 6
+    np_arr = np.array(array)
+    num_arr = num.array(array)
+    out_np = out_num = np.zeros(np_arr.shape)
```
Comment (bryevdv) [inline tests/integration/test_clip.py:193]: same comment as above

Comment (yimoj) [reply tests/integration/test_clip.py:193]: Definitely my mistake, and `out_num` in this one is not even meant to be a np array. Fixed.

File: tests/integration/test_clip.py [hunk 3]
```diff
@@ -146,7 +146,8 @@ def test_out_np_array():
     amax = 6
     np_arr = np.array(array)
     num_arr = num.array(array)
-    out_np = out_num = np.zeros(np_arr.shape)
+    out_np = np.zeros(np_arr.shape)
+    out_num = np.zeros(np_arr.shape)
```
Comment (bryevdv) [inline tests/integration/test_clip.py:150]: suggest `np.empty` is all that is required for out-arrays

**Review by bryevdv — APPROVED** [review]

**Pull request merged**